js/actions/index.js

```diff
@@ -23,22 +23,27 @@ export const ENABLE_ACTIVITY_FEEDBACK = 'ENABLE_ACTIVITY_FEEDBACK'
 // REQUEST_DATA action will be processed by the reducer immediately.
 // See: api-middleware.js
 function requestData() {
-  return {
-    type: REQUEST_DATA,
-    callAPI: {
-      type: 'fetchReportData',
-      successAction: receiveData,
-      errorAction: receiveError
-    }
+  return (dispatch, getState) => {
+    dispatch({
+      type: REQUEST_DATA,
+      callAPI: {
+        type: 'fetchReportData',
+        successAction: receiveData,
+        errorAction: receiveError
+      }
+    });
```

Comment:
Is that necessary? It's been a while since I used Redux last time, but I think explicit dispatch call (or thunk middleware) is useful either when you have some async operation, multiple actions dispatched in one function, or you need to use `getState`. This action looks kind of straightforward.

Reply:
You are right, this is not necessary.  I am not sure what happened here.  I might have just been going for consistency.